File: src-tauri/src/api/settings_api.rs
use sea_orm::DbConn;
use tauri::State;
use tokio::sync::Mutex;

use crate::{
    dto::update_settings_dto::UpdateSettingsRequest, model::settings::Settings,
    service::settings_service,
};

#[tauri::command]
pub async fn get_settings() -> Result<Settings, ()> {
    Ok(settings_service::get_settings())
}

#[tauri::command]
pub async fn update_settings(
    db_conn: State<'_, Mutex<DbConn>>,
    new_settings: UpdateSettingsRequest,
) -> Result<(), String> {
    settings_service::update_settings(new_settings, &db_conn).await;
    Ok(())
}

File: src/api/settingsApi.ts
import { invoke } from "@tauri-apps/api/core";
import Settings from "../type/backend/model/settings";
import UpdateSettingsRequest from "../type/backend/dto/updateSettingsRequestDto";

export function getSettings(): Promise<Settings> {
	return invoke("get_settings");
}

export function updateSettings(
	updateSettingsRequest: UpdateSettingsRequest,
): Promise<void> {
	return invoke("update_settings", {
		newSettings: updateSettingsRequest,
	});
}
